### DISCUSSION TITLE
How to create css class for Blazorise compoments?

### DISCUSSION THREAD

**Discussion by ghost** [Megabit/Blazorise#3358] [Q&A] [answered]
I have a lot of cards and want to make them looking unique with having all the same margin.
I could define each card with
`<Card Margin="Margin.Is2">`
but if I want to change the margin to e.g. ls4 I would have to change each card.

Can I write a class in my css file, e.g. looking like
`.MyCard {
    Margin: Margin.Is2;
}`
and apply it to all my Cards with
`<Card Class="MyCard">`
?

The problem is that `Margin: Margin.ls2;` is no correct css syntax.

Thanks for any help and sorry, but I'm a newbie to Blazorise and not very experienced with web programming yet.

**Comment by stsrki** [comment, marked as answer]
You can make static variable in a helper class, eg

```cshtml
static class Helpers
{
  public static readonly IFluentSpacing CardMargin = Margin.Is2;
}
```

and use it `<Card Margin="Helpers.CardMargin">`. It is now controlled from the same location.

Otherwise, you can also use the custom class with custom CSS, `<Card Class="MyCard">`

```css
.MyCard
{
  margin: 1.25rem;
}
```

**Reply by ghost** [reply]
I chose the second way (the standard margin), since there probably will be more than just setting the margin, so this is easier.
But the way with the static variable is a really good advice. I wasn't aware that access it directly in the Card-Tag is possible.
Thank's a lot :-))

**Comment by ghost** [comment]
Sorry, but an additional question concerning this arose:
How can I find out how Blazorise maps the (better: some) properties of it's components to standard (or bootstrap) css?

Example: My TextEdits are
`<TextEdit Size="Size.Small"></TextEdit>`
I want to css my TextEdits with a appropriate element in my styles file (like mentioned above in the margin example). For this, I searched the source of Blazorise (the example app) and there the TextEdit component and it's base classes. Finally I ended stucked somewhere between ClassBuilder, DirtyClasses() and blazorise.bootstrap.css 

Is there a way to get further in understanding this?

-----
Another idea for a workaround: Would it make sense to write my own blazor component "MyTextEdit" which derives from Blazorise.TextEdit and set there (in the constructor) the Size to Size.Small?

**Comment by stsrki** [comment]
We use ClassBuilder to append class names based on the current component conditions. `DirtyClasses()` is there so that ClassBuilder can be cleared once something changes, otherwise, cached class names are used.

`blazorise.bootstrap.css` comes to plan in situations where we want to implement some custom features that are not present in the native Bootstrap CSS. This we try to use as little as possible, and only where it is most needed.

---

You could inherit from `TextEdit` to build `MyTextEdit`, but then you must take care that is properly registered with Dependency injection and that it properly handles all the required events and validations.

**Reply by ghost** [reply]
Slowly...the fog goes away...
Thank you, again (thumbs up)
